packages/webapp/src/inductions/components/induction-journeys/invitee-journey.tsx

```diff
@@ -0,0 +1,156 @@
+        return renderProfileStep();
+    }
+
+    switch (inductionStatus) {
+        case InductionStatus.PendingProfile:
+            return renderProfileStep();
+        case InductionStatus.PendingCeremonyVideo: // not possible in Genesis mode
+            return (
+                <Container
+                    step={InductionStepInvitee.PendingVideoAndEndorsements}
+                    memberPreview={memberData}
+                >
+                    <WaitingForVideo induction={induction} />
+                </Container>
+            );
+        case InductionStatus.PendingEndorsement: // not possible in Genesis mode
+            return (
+                <Container
+                    step={InductionStepInvitee.PendingVideoAndEndorsements}
+                    memberPreview={memberData}
+                >
+                    <Heading size={1} className="mb-2">
+                        Endorsements
+                    </Heading>
+                    <InductionExpiresIn induction={induction} />
+                    <EndorsementsStatus endorsements={endorsements} />
+                    <div className="space-y-3">
+                        <Text>To continue, all witnesses must endorse.</Text>
+                        <Text>
+                            Now is a good time to review your profile
+                            information below. If anything needs to be
+                            corrected,{" "}
+                            <Link onClick={() => setIsReviewingProfile(true)}>
+                                click here to make those adjustments.
+                            </Link>{" "}
+                            Keep in mind that any modifications to your profile
+                            will reset any endorsements.
+                        </Text>
+                    </div>
+                </Container>
+            );
+        case InductionStatus.PendingDonation:
+            return (
+                <Container
+                    step={
+                        isCommunityActive
+                            ? InductionStepInvitee.Donate
+                            : InductionStepGenesis.Donate
+                    }
+                    memberPreview={memberData}
+                >
+                    <Heading size={1} className="mb-2">
+                        Pending donation
+                    </Heading>
+                    <InductionExpiresIn induction={induction} />
+                    {isCommunityActive && (
+                        <EndorsementsStatus endorsements={endorsements} />
+                    )}
+                    <InductionDonateForm
+                        induction={induction}
+                        isCommunityActive={isCommunityActive}
+                        setIsReviewingProfile={setIsReviewingProfile}
+                    />
+                </Container>
+            );
+        default:
+            return <p>Unknown error</p>;
+    }
+};
+
+export default InviteeJourney;
```

Comment:
We could get it even simpler and readable with extracting the branching to it's own components, something like this (need to replace the any's though): 

```tsx
// ...

export const InviteeJourney = ({
    endorsements,
    induction,
    inductionStatus,
}: Props) => {
    const [submittedProfile, setSubmittedProfile] = useState(false);
    const [isReviewingProfile, setIsReviewingProfile] = useState(false);

    if (submittedProfile) {
        return <SubmittedProfileStep />;
    }

    if (isReviewingProfile) {
        return (
            <ProfileStep
                induction={induction}
                isReviewingProfile={isReviewingProfile}
                setSubmittedProfile={setSubmittedProfile}
            />
        );
    }

    switch (inductionStatus) {
        case InductionStatus.PendingProfile:
            return (
                <ProfileStep
                    induction={induction}
                    isReviewingProfile={isReviewingProfile}
                    setSubmittedProfile={setSubmittedProfile}
                />
            );
        case InductionStatus.PendingCeremonyVideo: // not possible in Genesis mode
            return <PendingCeremonyVideoStep induction={induction} />;
        case InductionStatus.PendingEndorsement: // not possible in Genesis mode
            return (
                <PendingEndorsementStep
                    induction={induction}
                    endorsements={endorsements}
                    setIsReviewingProfile={setIsReviewingProfile}
                />
            );
        case InductionStatus.PendingDonation:
            return (
                <PendingDonateStep
                    induction={induction}
                    endorsements={endorsements}
                    setIsReviewingProfile={setIsReviewingProfile}
                />
            );
        default:
            return <p>Unknown error</p>;
    }
};

export default InviteeJourney;

const SubmittedProfileStep = () => {
    const { data: isCommunityActive } = useIsCommunityActive();
    return (
        <Container
            step={
                isCommunityActive
                    ? InductionStepInvitee.PendingVideoAndEndorsements
                    : InductionStepGenesis.Donate
            }
        >
            <InductionProfileSubmitConfirmation
                isCommunityActive={isCommunityActive}
            />
        </Container>
    );
};

const ProfileStep = ({
    induction,
    isReviewingProfile,
    setSubmittedProfile,
}: any) => {
    const { data: isCommunityActive } = useIsCommunityActive();
    return (
        <Container
            step={
                isCommunityActive
                    ? InductionStepInvitee.Profile
                    : InductionStepGenesis.Profile
            }
        >
            <InductionProfileFormContainer
                induction={induction}
                isReviewingProfile={isReviewingProfile}
                setSubmittedProfile={setSubmittedProfile}
            />
        </Container>
    );
};

const PendingCeremonyVideoStep = ({ memberData, induction }: any) => (
    <Container
        step={InductionStepInvitee.PendingVideoAndEndorsements}
        memberPreview={memberData}
    >
        <WaitingForVideo induction={induction} />
    </Container>
);

const PendingEndorsementStep = ({
    induction,
    endorsements,
    setIsReviewingProfile,
}: any) => {
    const memberData = convertPendingProfileToMemberData(induction);

    return (
        <Container
            step={InductionStepInvitee.PendingVideoAndEndorsements}
            memberPreview={memberData}
        >
            <Heading size={1} className="mb-2">
                Endorsements
            </Heading>
            <InductionExpiresIn induction={induction} />
            <EndorsementsStatus endorsements={endorsements} />
            <div className="space-y-3">
                <Text>To continue, all witnesses must endorse.</Text>
                <Text>
                    Now is a good time to review your profile information below.
                    If anything needs to be corrected,{" "}
                    <Link onClick={() => setIsReviewingProfile(true)}>
                        click here to make those adjustments.
                    </Link>{" "}
                    Keep in mind that any modifications to your profile will
                    reset any endorsements.
                </Text>
            </div>
        </Container>
    );
};

const PendingDonateStep = ({
    induction,
    endorsements,
    setIsReviewingProfile,
}: any) => {
    const memberData = convertPendingProfileToMemberData(induction);
    const { data: isCommunityActive } = useIsCommunityActive();

    return (
        <Container
            step={
                isCommunityActive
                    ? InductionStepInvitee.Donate
                    : InductionStepGenesis.Donate
            }
            memberPreview={memberData}
        >
            <Heading size={1} className="mb-2">
                Pending donation
            </Heading>
            <InductionExpiresIn induction={induction} />
            {isCommunityActive && (
                <EndorsementsStatus endorsements={endorsements} />
            )}
            <InductionDonateForm
                induction={induction}
                isCommunityActive={isCommunityActive}
                setIsReviewingProfile={setIsReviewingProfile}
            />
        </Container>
    );
};
```

Reply:
Oooh, I _like_ it!